www/docs/cookbooks/sign-and-notarizing-osx.md

```diff
@@ -0,0 +1,270 @@
+          }
+          source = ["dist/macos_{{.Target}}/{{.Name}}"]
+          sign {
+            application_identity = "{{.Env.APPLE_APPLICATION_IDENTITY}}"
+          }
+          EOF
+          '
+        - "gon 'dist/macos_{{.Target}}/gon.hcl'"
+```
+
+Result: the `dist/macos_*/myapp` will be signed.
+
+```
+$ codesign --verify --deep --verbose dist/macos_darwin_arm64/myapp
+dist/macos_darwin_arm64/myapp: valid on disk
+dist/macos_darwin_arm64/myapp: satisfies its Designated Requirement
+```
+
+
+## Configuring Notarization
+
+Notarizing takes our archived (with a signed binary inside) and submits it to Apple. This is an online activity than can take meaningful time but it necessary for production releases to avoid security prompts.
+
+Here we are doing a very similar behavior as we did during the Build step, where we define a hon config block on the fly and then execute.
+
+We must be sure to define an archive ID, so we when we go to Release later, Goreleaser will pick the correct files. Here we do a specific notarize step _replacing_ the original artifact in-place. One could generate a separate archive filename, just be sure not to Release it.
+
+**Important**:
+  - zip, pkg, dmg, app format is required for notarization.
+
+```yaml
+# .goreleaser.yml
+
+archives:
+  - format: zip
+    builds: [default]
+    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
+    format_overrides:
+      - goos: linux
+        format: tar.gz
+  - id: macos
+    builds: [macos]
+    format: zip
+    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
+
+signs:
+  - id: mac-notarize
+    artifacts: archive
+    ids: [macos]
+    signature: "${artifact}"
+    output: true
+    cmd: sh
+    args:
+      - "-c"
+      - |-
+        cat >"dist/gon.notarize.hcl" <<EOF
+        apple_id {
+          username = "{{.Env.APPLE_ID_USERNAME}}"
+          password = "{{.Env.APPLE_ID_PASSWORD}}"
+        }
+        notarize {
+          path = "${artifact}"
+          bundle_id = "{{.Env.BUNDLE_ID}}"
+        }
+        EOF
+        gon "dist/gon.notarize.hcl"
+```
+
+Result: notarized artifacts.
+FIXME: We know it notarized correctly, why is our archive being rejected by spctl?
+
+```
+$ spctl --assess -v dist/myapp_1.0.0_darwin_arm64.zip
+dist/myapp_1.0.0_darwin_arm64.zip: rejected
+source=no usable signature
+```
+
+## Releasing
+
+Since during the Sign step we generated a new set of notarized artifacts, GoReleaser now thinks there are a double-set of OSX related artifacts (ones from the Artifacts step, ones from Sign step). When attempting to Release, it will try to release both sets.
+
+If we've used the same name like in this example our publishing services may complain about overwriting existing artifacts. Instead we explictly inform goreleaser which artifacts to upload. Excluding the `macos` artifacts ID to prevent the duplication error.
+
+```yaml
+release:
+  ids: [default, mac-notarize]
+```
```

Comment:
this can probably be improved in goreleaser

Reply:
I think so, my releases failed because GitLab wouldn't take the duplicate artifact name/file uploads. Explicitly selecting one was the only recourse. It was just confusing to identify because I was re-using the archive's name.